be/src/vec/exprs/vruntimefilter_wrapper.cpp

```diff
@@ -0,0 +1,108 @@
+
+VRuntimeFilterWrapper::VRuntimeFilterWrapper(const TExprNode& node, VExpr* impl)
+        : VExpr(node), _impl(impl), _always_true(false), _filtered_rows(0), _scan_rows(0) {}
+
+VRuntimeFilterWrapper::VRuntimeFilterWrapper(const VRuntimeFilterWrapper& vexpr)
+        : VExpr(vexpr),
+          _impl(vexpr._impl),
+          _always_true(vexpr._always_true),
+          _filtered_rows(vexpr._filtered_rows.load()),
+          _scan_rows(vexpr._scan_rows.load()) {}
+
+Status VRuntimeFilterWrapper::prepare(RuntimeState* state, const RowDescriptor& desc,
+                                      VExprContext* context) {
+    return _impl->prepare(state, desc, context);
+}
+
+Status VRuntimeFilterWrapper::open(RuntimeState* state, VExprContext* context,
+                                   FunctionContext::FunctionStateScope scope) {
+    return _impl->open(state, context, scope);
+}
+
+void VRuntimeFilterWrapper::close(RuntimeState* state, VExprContext* context,
+                                  FunctionContext::FunctionStateScope scope) {
+    _impl->close(state, context, scope);
+}
+
+bool VRuntimeFilterWrapper::is_constant() const {
+    return _impl->is_constant();
+}
+
+Status VRuntimeFilterWrapper::execute(VExprContext* context, Block* block, int* result_column_id) {
+    if (_always_true) {
+        auto res_data_column = ColumnVector<UInt8>::create(block->rows());
+        size_t num_columns_without_result = block->columns();
+        size_t sz = block->rows();
+        res_data_column->resize(sz);
+        res_data_column->reserve(sz);
+        auto ptr = ((ColumnVector<UInt8>*)res_data_column.get())->get_data().data();
+        for (size_t i = 0; i < sz; i++) {
+            ptr[i] = 1;
+        }
+        if (_data_type->is_nullable()) {
+            auto null_map = ColumnVector<UInt8>::create(block->rows(), 0);
+            block->insert({ColumnNullable::create(std::move(res_data_column), std::move(null_map)),
+                           _data_type, expr_name()});
+        } else {
+            block->insert({std::move(res_data_column), _data_type, expr_name()});
+        }
+        *result_column_id = num_columns_without_result;
+        return Status::OK();
+    } else {
+        _scan_rows += block->rows();
+        auto status = _impl->execute(context, block, result_column_id);
+        auto* data =
+                ((ColumnVector<UInt8>*)block->get_columns_with_type_and_name()[*result_column_id]
+                         .column.get())
+                        ->get_data()
+                        .data();
+        _filtered_rows +=
+                doris::simd::count_zero_num(reinterpret_cast<const int8_t*>(data), block->rows());
+
+        if ((!_has_calculate_filter) && (_scan_rows > 0) && (_scan_rows.load() >= _loop_size)) {
+            double rate = (double)_filtered_rows / _scan_rows;
+            if (rate < _expect_filter_rate) {
+                _always_true = true;
+            }
+            _has_calculate_filter = true;
```

Comment:
why we need `_has_calculate_filter`? seems `_scan_rows > 0` is enough

Reply:
`(_scan_rows > 0) && (_scan_rows.load() >= _loop_size`
Is x somewhat redundant?

Reply:
> why we need `_has_calculate_filter`? seems `_scan_rows > 0` is e

I just want to use the origin logics in `exprs/bloomfilter_predicate.cpp` here. I think using `_has_calculate_filter` will ensure we only calculate filter rate once when scan rows count is large than `_loop_size`